core/src/main/java/org/infinispan/container/impl/EntryFactoryImpl.java

```diff
@@ -101,25 +177,48 @@ public void wrapEntryForWriting(InvocationContext ctx, Object key, int segment,
             log.tracef("Updated context entry %s -> %s", contextEntry, mvccEntry);
       } else {
          // Not in the context yet.
-         CacheEntry cacheEntry = getFromContainerForWrite(key, segment, isOwner);
-         if (cacheEntry == null) {
-            return;
-         }
-         MVCCEntry mvccEntry = createWrappedEntry(key, cacheEntry);
-         if (cacheEntry.isNull()) {
-            mvccEntry.setCreated(true);
+         InternalCacheEntry ice = getFromContainer(key, segment);
+         if (isOwner) {
+            if (ice == null) {
+               addWriteEntryToContext(ctx, NullCacheEntry.getInstance(), key, isRead);
+            } else {
+               CompletionStage<Boolean> expiredStage = handlePossibleExpiredEntry(ice, true);
+               if (expiredStage != null) {
+                  if (CompletionStages.isCompletedSuccessfully(expiredStage)) {
+                     Boolean expired = CompletionStages.join(expiredStage);
+                     handleWriteExpiredEntryContextAddition(expired, ctx, ice, key, isOwner);
```

Comment:
should be `isRead` instead of `isOnwer`. Same typo below.

Reply:
I copy/pasted from the read one, heh.

Reply:
fixed.